### PR TITLE
test(scorecard): retry on cryostat update conflict

### PR DESCRIPTION
**Pull request by mwangggg**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits): `git commit -S -m "YOUR_COMMIT_MESSAGE"`
_______________________________________________

Fixes: #757

### CODE REVIEW

File: internal/test/scorecard/tests.go [hunk 1]
```diff
@@ -101,28 +99,6 @@ func CryostatConfigChangeTest(bundle *apimanifests.Bundle, namespace string, ope
...
-					Requests: corev1.ResourceList{
-						corev1.ResourceStorage: resource.MustParse("1Gi"),
-					},
-				},
-			},
-		},
-	}
-
-	// Wait for redeployment of Cryostat CR
 	err = updateAndWaitTillCryostatAvailable(cr, tr)
```
Comment (tthvo) [inline internal/test/scorecard/tests.go:102]: I am seeing this failure. The `cr` here is not updated and is actually just an empty struct, thus the `status.ApplicationURL` is `""`

```
Image:      quay.io/thvo/cryostat-operator-scorecard:2.5.0-20240326201336
Entrypoint: [cryostat-scorecard-tests cryostat-config-change]
Labels:
	"suite":"cryostat"
	"test":"cryostat-config-change"
Results:
	Name: cryostat-config-change
	State: fail

	Errors:
		failed to reach the application: Get "health": unsupported protocol scheme ""
	Log:
		deployment cryostat-config-change is not yet found
		deployment cryostat-config-change is not yet found
		deployment cryostat-config-change is not yet found
		deployment cryostat-config-change is not yet found
		deployment cryostat-config-change is not yet found
		deployment cryostat-config-change is not yet found
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is not yet available
		deployment cryostat-config-change is available
		application is available at https://testing.cryostat
		Cryostat deployment is still updating. Storage: &EmptyDirVolumeSource{Medium:,SizeLimit:0,}
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 new replicas have been updated... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 new replicas have been updated... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		Waiting for deployment cryostat-config-change rollout to finish: 0 out of 1 updated replicas are available... 
		deployment cryostat-config-change successfully rolled out
		Cryostat deployhttps://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntaxment has successfully updated with new spec template
```

**Comment by tthvo** [comment]
The test failure seems to be the same as https://github.com/cryostatio/cryostat-operator/pull/774#discussion_r1540106491. Maybe rebase lost the changes?

File: internal/test/scorecard/common_utils.go [hunk 2]
```diff
@@ -511,11 +513,33 @@ func (r *TestResources) sendHealthRequest(base *url.URL, healthCheck func(resp *
...
 
-func (r *TestResources) updateAndWaitTillCryostatAvailable(cr *operatorv1beta1.Cryostat) error {
-	cr, err := r.Client.OperatorCRDs().Cryostats(cr.Namespace).Update(context.Background(), cr)
-	if err != nil {
-		r.Log += fmt.Sprintf("failed to update Cryostat CR: %s", err.Error())
+func (r *TestResources) updateAndWaitTillCryostatAvailable(cr *operatorv1beta1.Cryostat) (*operatorv1beta1.Cryostat, error) {
+	ctx := context.Background()
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cr, err := r.Client.OperatorCRDs().Cryostats(cr.Namespace).Get(ctx, cr.Name)
```
Comment (tthvo) [inline internal/test/scorecard/common_utils.go:520]: This line declares another `cr` variable within the scope of retry func. Thus, the returned `cr` still holds pointer to the empty CR struct.

Comment (tthvo) [reply internal/test/scorecard/common_utils.go:520]: A simple fix can be:

```suggestion
        var err error
		cr, err = r.Client.OperatorCRDs().Cryostats(cr.Namespace).Get(ctx, cr.Name)
```

**Review by ebaron — DISMISSED** [review]
Looks good, thanks @mwangggg! Just needs a rebase with `make scorecard-build MANIFEST_PUSH=false bundle`.

**Review by ebaron — APPROVED** [review]

**Comment by ebaron** [comment]
@Mergifyio backport cryostat3

**Pull request merged**